ccl/serverccl: TestTableAndDatabaseDetailsAndStats failed

ccl/serverccl.TestTableAndDatabaseDetailsAndStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12186348?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12186348?buildTab=artifacts#/) on release-23.2 @ [87a94f6ac59d4e01c822bb2128db858176600dd6](https://github.com/cockroachdb/cockroach/commits/87a94f6ac59d4e01c822bb2128db858176600dd6):


```
=== RUN   TestTableAndDatabaseDetailsAndStats
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/c101b7a464a1afc1f5af0cd85792187e/logTestTableAndDatabaseDetailsAndStats1033658230
    test_log_scope.go:81: use -show-logs to present logs inline
    admin_test.go:240: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/serverccl/admin_test.go:240
        	Error:      	"0" is not greater than "0"
        	Test:       	TestTableAndDatabaseDetailsAndStats
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/c101b7a464a1afc1f5af0cd85792187e/logTestTableAndDatabaseDetailsAndStats1033658230
--- FAIL: TestTableAndDatabaseDetailsAndStats (9.64s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTableAndDatabaseDetailsAndStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32389

--- a/pkg/ccl/serverccl/admin_test.go
+++ b/pkg/ccl/serverccl/admin_test.go
@@ -233,9 +233,16 @@ func TestTableAndDatabaseDetailsAndStats(t *testing.T) {
 	require.Greater(t, tableStatsResp.Stats.LiveBytes, int64(0))
 
 	// TableDetails
-	tableDetailsResp := &serverpb.TableDetailsResponse{}
-	err = getAdminJSONProto(st, "databases/defaultdb/tables/public.test", tableDetailsResp)
-	require.NoError(t, err)
-
-	require.Greater(t, tableDetailsResp.DataLiveBytes, int64(0))
+	// Call to endpoint is wrapped with retry logic to potentially avoid flakiness of
+	// returned results (issue #112387).
+	testutils.SucceedsSoon(t, func() error {
+		tableDetailsResp := &serverpb.TableDetailsResponse{}
+		if err := getAdminJSONProto(st, "databases/defaultdb/tables/public.test", tableDetailsResp); err != nil {
+			return err
+		}
+		if tableDetailsResp.DataLiveBytes <= 0 {
+			return fmt.Errorf("expected DataLiveBytes to be greater than 0 but got %d", tableDetailsResp.DataLiveBytes)
+		}
+		return nil
+	})
 }

